### PR TITLE
refactor config

### PR DESCRIPTION
**Pull request by joshklop**
Fixes: #230 #300 

## Changes
- Process configurations from various sources (in order of precedence): CLI parameters, environment variables, config file entries, and default values.
- Use consistent naming conventions regardless of the configuration source
- Never modify the user's environment except for creating database files in locations explicitly provided by the user or in clearly defined default locations.

## Types of changes

What types of changes does your code introduce?

- [X] New feature (more CLI flags and corresponding environment variables)
- [X] Breaking change (previous configurations will break)
- [X] Refactoring

## Testing
**Requires testing**

- [ ] Yes
- [X] No (currently we do not test anything in `cmd/juno`)

**In case you checked yes, did you write tests??**

- [ ] Yes
- [X] No

**Note**: now that `main.go` is getting more modular, I think we can add tests for most of it in the near future.

### CODE REVIEW

**Review by adriantpaez — APPROVED** [review]
LGTM

File: internal/config/config.go [hunk 1]
```diff
@@ -4,175 +4,100 @@ package config
...
+// empty string and error otherwise.
+// It is analagous to os.UserConfigDir.
+func UserDataDir() (string, error) {
 	// notest
-	d, err := os.UserConfigDir()
-	if err != nil {
-		Logger.Fatal("Could not get config directory.")
-	}
-	Dir = filepath.Join(d, "juno")
+	junoDir := "juno"
```
Comment (tshakalekholoane) [inline internal/config/config.go:68]: This doesn't look to ever be mutated afterwards to let's initialise it as a `const` for clarity.

**Review by tshakalekholoane — CHANGES_REQUESTED** [review]

**Comment by D-DePablos** [comment]
If we expect the user to use the --starknet-enable flag, it is not routed with make all `starknet enable`. 

Since our [docs](https://gojuno.xyz/docs/intro) indicate the user to `make run`, or `make all`, this would not run correctly as the docs stand. 

image

**Review by tshakalekholoane — APPROVED** [review]
LGTM.

**Comment by stdevMac** [comment]
Can we add more details around Environment vars and params on the docs? @joshklop Thanks in advance

**Review by stdevMac — APPROVED** [review]

**Pull request merged**